activerecord/lib/active_record/associations/alias_tracker.rb

```diff
@@ -51,19 +52,20 @@ def self.initial_count_for(connection, name, table_joins)
       end
 
       # table_joins is an array of arel joins which might conflict with the aliases we assign here
-      def initialize(connection, aliases)
+      def initialize(connection, aliases, type_caster)
         @aliases    = aliases
         @connection = connection
+        @type_caster = type_caster
       end
 
-      def aliased_table_for(table_name, aliased_name, **table_options)
+      def aliased_table_for(table_name, aliased_name)
         if aliases[table_name].zero?
           # If it's zero, we can have our table_name
           aliases[table_name] = 1
-          Arel::Table.new(table_name, table_options)
+          Arel::Table.new(table_name, type_caster: @type_caster)
         else
           # Otherwise, we need to use an alias
-          aliased_name = connection.table_alias_for(aliased_name)
+          aliased_name = @connection.table_alias_for(aliased_name)
```

Comment:
I don't feel that strongly one way or another, but the reasoning was `@connection` was already available so the attr_reader was unnecessary. I'll defer to @tenderlove in case he has strong feelings about protected attr_reader vs the ivar.